### PR TITLE
Update xeus to 5.1.0 and xeus-zmq to 3.0.0

### PR DESCRIPTION
**Pull request by KGB99**
This PR is to update xeus from 3.0.5 to 5.1.0 and xeus-zmq from 1.0.2 to 3.0.0

### CODE REVIEW

**Comment by KGB99** [comment]
This PR is to update xeus from 3.0.5 to 5.1.0 and xeus-zmq from 1.0.2 to 3.0.0

File: ci/environment_linux_llvm.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ dependencies:
   - xeus=5.1.0
   - xeus-zmq=3.0.0
   - xtl
-  - nlohmann_json=3.11.2
+  - nlohmann_json>=3.11.2
```
Comment (certik) [inline ci/environment_linux_llvm.yml:11]: Set the exact version. We set this version due to a build failure with a newer version.

So if you need to update to a newer version, pick a version that works and set it exactly here.

File: src/lfortran/fortran_kernel.cpp [hunk 2]
```diff
@@ -17,8 +17,8 @@
 #include <xeus/xinterpreter.hpp>
 #include <xeus/xkernel.hpp>
 #include <xeus/xkernel_configuration.hpp>
-#include <xeus-zmq/xserver_zmq.hpp>
-#include "xeus-zmq/xserver_shell_main.hpp"
+#include "xeus-zmq/xzmq_context.hpp"
+#include <xeus-zmq/xserver_zmq_split.hpp>
```
Comment (certik) [inline src/lfortran/fortran_kernel.cpp:21]: Why not to use `<` everywhere for consistency?

**Review by certik — APPROVED** [review]
I think this looks great, thanks for getting it done. I left a minor comment that can be fixed in  a subsequent PRs.

**Comment by certik** [comment]
The Windows failure will be fixed by https://github.com/lfortran/lfortran/pull/4310.

**Review by certik — APPROVED** [review]

**Pull request merged**